contribs/desync-resources.py

```diff
@@ -0,0 +1,185 @@
+
+    if not rcl and not users:
+        sys.exit(0)
+
+    result = io.StringIO()
+    fieldnames = ['uuid', 'type', 'name']
+    writer = csv.DictWriter(result, fieldnames=fieldnames)
+    writer.writeheader()
+    for row in rcl + users:
+        writer.writerow(row)
+
+    if args.output:
+        with open(args.output, 'w') as fobj:
+            result.seek(0)
+            shutil.copyfileobj(result, fobj)
+    else:
+        print(result.getvalue())
+
+    result.close()
+    sys.exit(1)
+
+
+def detect_desync_users(auth_client, confd_client, force_synchro):
+    result = []
+    response = auth_client.users.list(recurse=True)
+    users = {user['uuid']: user for user in response['items']}
+
+    confd_users = confd_client.users.list()['items']
+    for confd_user in confd_users:
+        user = users.get(confd_user['uuid'])
+        if not user:
+            result.append({
+                'uuid': confd_user['uuid'],
+                'type': 'user',
+                'name': '',
+            })
+            if force_synchro:
+                confd_client.users.delete(confd_user['uuid'])
+    return result
+
+
+def detect_desync_rcl(auth_client, confd_client):
+    result = []
+    response = auth_client.tenants.list()
+    tenants = {tenant['uuid']: tenant for tenant in response['items']}
+
+    resellers = confd_client.resellers.list()['items']
+    for reseller in resellers:
+        tenant = tenants.get(reseller['uuid'])
+        if not tenant:
+            result.append({
+                'uuid': reseller['uuid'],
+                'type': 'reseller',
+                'name': reseller['name'],
+            })
+
+    customers = confd_client.customers.list()['items']
+    for customer in customers:
+        tenant = tenants.get(customer['uuid'])
+        if not tenant:
+            result.append({
+                'uuid': customer['uuid'],
+                'type': 'customer',
+                'name': customer['name'],
+            })
+
+    locations = confd_client.locations.list()['items']
+    for location in locations:
+        tenant = tenants.get(location['uuid'])
+        if not tenant:
+            result.append({
+                'uuid': location['uuid'],
+                'type': 'location',
+                'name': location['name'],
+            })
+    return result
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="""
+        Script to detect desynchronization between wazo-auth and nestbox-confd.
+
+        Only users can be removed automatically from nestbox-confd because impacts are mitigate
+        All other destructive operation must be done manually
+        """
+    )
+    parser.add_argument(
+        'host',
+        help="The nestbox hostname",
+    )
+    parser.add_argument(
+        '-u',
+        '--username',
+        required=True,
+    )
+    parser.add_argument(
+        '-p',
+        '--password',
```

Comment:
Password should be prompted as secret, or taken from an env variable.

Reply:
Fixed